crates/bevy_window/src/raw_handle.rs

```diff
@@ -5,6 +5,38 @@ use raw_window_handle::{
     DisplayHandle, HandleError, HasDisplayHandle, HasWindowHandle, RawDisplayHandle,
     RawWindowHandle, WindowHandle,
 };
+use std::{any::Any, marker::PhantomData, ops::Deref, sync::Arc};
+
+/// A wrapper over a window.
+///
+/// This allows us to extend the lifetime of the window, so it doesn't get eagerly dropped while a
+/// pipelined renderer still has frames in flight that need to draw to it.
+///
+/// This is achieved by storing a shared reference to the window in the [`RawHandleWrapper`],
+/// which gets picked up by the renderer during extraction.
+#[derive(Debug)]
+pub struct WindowWrapper<W> {
+    reference: Arc<dyn Any + Send + Sync>,
```

Comment:
Why does this need to be type erased instead of storing a Arc<W>?

Reply:
This is so that we can later store store this Arc in the `RawHandleWrapper` without static typing:
https://github.com/bevyengine/bevy/blob/5a013571b43f8072a92cf4f4ebc4a7f1c0cfd17b/crates/bevy_window/src/raw_handle.rs#L47-L48